### PR TITLE
Refactor and add ut for  ClassUtils

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- add unit tests for `ClassUtils`
- refactor `ClassUtils.createInstance` method with DynConstructors
- move `classIsLoadable` method to `ReflectUtils.isClassLoadable`, refeactor to use DynClasses

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala [hunk 1]
```diff
@@ -110,7 +110,7 @@ case class EngineTab(
     // but if use sbt to build for testing, e.g. docker image, it still uses vanilla jetty classes.
     val shaded = "org.sparkproject.jetty.servlet.ServletContextHandler"
     val vanilla = "org.eclipse.jetty.servlet.ServletContextHandler"
-    if (ClassUtils.classIsLoadable(shaded)) {
+    if (ReflectUtils.isClassLoadable(shaded)) {
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala:113]: maybe we can inline this method with the following code as replacement
```
    DynClasses.builder()
      .impl("org.sparkproject.jetty.servlet.ServletContextHandler")
      .impl("org.eclipse.jetty.servlet.ServletContextHandler")
      .build()
```

Comment (bowenliang123) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala:113]: It's done in #4879 , by using chaining builders.
https://github.com/apache/kyuubi/pull/4879/files#diff-9a53f1a1af5e3f8d7bebf106722791153acce818fc25e6c61c479f628e9962bcR112

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala:113]: it's not enough, let's do it in this one.

- use the chained builder
- inline the code and remove method `loadSparkServletContextHandler`
- rewrite the code behind invoking of `loadSparkServletContextHandler`

Comment (bowenliang123) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala]: Addressed.

**Review by pan3793 — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala [hunk 2]
```diff
@@ -63,7 +63,13 @@ case class EngineTab(
   sparkUI.foreach { ui =>
     try {
       // Spark shade the jetty package so here we use reflection
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala:65]:
```suggestion
```

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala [hunk 3]
```diff
@@ -62,8 +62,13 @@ case class EngineTab(
 
   sparkUI.foreach { ui =>
     try {
-      // Spark shade the jetty package so here we use reflection
-      val sparkServletContextHandlerClz = loadSparkServletContextHandler
+      // [KYUUBI #3627]: the official spark release uses the shaded and relocated jetty classes,
+      // but if use sbt to build for testing, e.g. docker image, it still uses
+      // vanilla jetty classes.
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/ui/EngineTab.scala:67]:
```suggestion
      // [KYUUBI #3627]: the official spark release uses the shaded and relocated jetty classes,
      // but if we use sbt to build for testing, e.g. docker image, it still uses the vanilla
      // jetty classes.
```

**Comment by bowenliang123** [comment]
Can I move `isClassLoadable` method from `kyuubi-util-scala`'s `ReflectUtils` to `kyuubi-util`'s `DynClasses` as a static method for Java as well ? @pan3793

**Comment by pan3793** [comment]
> Can I move the `isClassLoadable` method from `kyuubi-util-scala`'s `ReflectUtils` to `kyuubi-util`'s `DynClasses` as a static method for Java as well ? @pan3793

It's fine if other places can benefit from it, but since Java does not support the default value, it should be broken into multi-methods

**Comment by bowenliang123** [comment]
> > Can I move the `isClassLoadable` method from `kyuubi-util-scala`'s `ReflectUtils` to `kyuubi-util`'s `DynClasses` as a static method for Java as well ? @pan3793
> 
> It's fine if other places can benefit from it, but since Java does not support the default value, it should be broken into multi-methods

OK, let's leave it in Scala module for now. As for Java, it's handy to use workaround `DynClasses.builder().loader(cl).impl(className).orNull().build() != null` for checking the class loadable.

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**